dags/atd_moped_components_to_agol.py

```diff
@@ -69,4 +69,4 @@
         mount_tmp_dir=False,
     )
 
-    t1
+    date_filter_arg >> t1
```

Comment:
This is a good change, establishing an order that these must be run in. Would you consider including the third task in here so that we are assured that there will always be populated secrets before starting the subsequent tasks?

Reply:
should we put `env_vars` before all of our tasks then? 

`date_filter_arg >> env_vars >> t1 >> t2` etc?

Reply:
this is a really good point! Really makes me wonder if we've seen the same type of error for 1PW secrets. I don't see any DAGs where we do this but it seems like we should, right? `get_env_vars_task` has a task decorator and also it is an async request so it seems even more susceptible to a race condition... 🤔 

I'm going to add it. 🙏

Reply:
i think we might be overlooking something. I'm looking at the graph from John's PR, and I'm seeing `env_vars` as a dependency without declaring it this way. https://github.com/cityofaustin/atd-airflow/pull/194.

I'm going to take a closer look. 🔍

Reply:
maybe because its passed into the Docker Operator whereas the date_filter_arg is in a string? I didnt think about this difference

Reply:
I recall that the new DAG format provides for a more "implicit" way of setting the order and dependencies. For example, before anything has been added, and it's just `date_filter_arg >> t1` is in the file, Airflow is able to figure this much about the DAG's graph:

![image](https://github.com/user-attachments/assets/f1daecdc-4a60-447d-ad6a-5be447f8a6b2)

Reply:
cool, thanks for pasting that in here, Frank! then, i'll keep this PR as-is since it would be a much larger change to update every DAG that uses `get_env_vars_task`. Re-requesting review 🙏

Reply:
I think this PR is a non-change under this assumption. Backing out the change of e4ededac2b51333d52c0b59922b3deb3e5201ac7 leaves Airflow reporting this graph for the DAG. I feel like I must be overlooking something - can you point me in the right direction?

![image](https://github.com/user-attachments/assets/6f403dc2-8e72-4e07-a91c-fdcb43ee9483)

Reply:
it's really confounding :/  there's a slack thread somewhere in which i naively declare this to be an Airflow bug 🤷 

the `moped_components_to_agol` operator is getting passed the string `"None"` in  place of `{date_filter_arg}`. i'm resorting to speculation in our duck-typed world, but the [`get_date_filter_arg` ](https://github.com/cityofaustin/atd-airflow/blob/824cea830ea52ca35411b72846aef883bfdbe6ed/dags/utils/knack.py#L8) function itself cannot return `None` or `"None"` as far as i can tell, but it could use another look.

for what it's worth this patch did seem to resolve this error for the signal status publisher. it's really confounding :/

Reply:
agreed! thanks for the discussion, y'all. I'm going to merge and pull the changes on p01. I'll monitor but also pretty confident this will fix it even if we don't fully understand it.

Reply:
fwiw here's what this DAG looks like in my local `production` branch — it's showing the missing dependency on `get_env_vars` 🤷 

<img width="470" alt="Screenshot 2024-07-16 at 11 04 22 AM" src="https://github.com/user-attachments/assets/ca38aa1d-44eb-4578-9619-98ef6015be21">